msteams-platform/bots/how-to/bot-configuration-experience.md

```diff
@@ -54,135 +65,196 @@ You must configure the `fetchTask` property under `bots.configuration` object in
+
+   ```javascript
+      async handleTeamsConfigFetch(_context, _configData) {
+         let response = {};
+         const adaptiveCard = CardFactory.adaptiveCard(this.adaptiveCardForContinue());
+         response = {
+            config: {
+            value: {
+               card: adaptiveCard,
+               height: 500,
+               width: 600,
+               title: 'test card',
+            },
+            type: 'continue',
+            },
+         };
+         return response;
+      }
+   ```
+
+   ---
+
+* `type: "auth"`: You can also request the user to authenticate as a response to `config/fetch` request. The `type: "auth"` configuration prompts the user to sign in through a specified URL, which must be linked to a valid authentication page that can be opened in a browser. Authentication is essential for scenarios where the bot requires the user to be authenticated. It ensures that the user’s identity is verified, maintaining security, and personalized experiences within the bot’s functionality.
+
+   > [!NOTE]
+   > For `type: "auth"` only third party authentication is supported. Single sign-on (SSO) isn't supported. For more information on third party authentication, see [add authentication.](../../messaging-extensions/how-to/add-authentication.md)
+
+   # [C#](#tab/teams-bot-sdk2)
+
+   [Sample code reference](https://github.com/OfficeDev/Microsoft-Teams-Samples/blob/main/samples/bot-configuration-app-auth/csharp/Bot%20configuration/Bots/TeamsBot.cs#L78)
+
+   ```csharp
+   protected override Task<ConfigResponseBase> OnTeamsConfigFetchAsync(ITurnContext<IInvokeActivity> turnContext, JObject configData, CancellationToken cancellationToken)
+   {
+      ConfigResponseBase response = new ConfigResponse<BotConfigAuth>
+   {
+         Config = new BotConfigAuth
+         {
+            SuggestedActions = new SuggestedActions
+            {
+               Actions = new List<CardAction>
+               {
+                     new CardAction
+                     {
+                        type: "openUrl",
+                        value: "https://example.com/auth",
+                        title: "Sign in to this app"
+                     }
+               }
+            },
+            Type = "auth"
+         }
+   };
+   ```
+
+   # [JavaScript](#tab/JS2)
+
+   [Sample code reference](https://github.com/OfficeDev/Microsoft-Teams-Samples/blob/main/samples/bot-configuration-app-auth/nodejs/teamsBot.js#L51)
+
+   ```javascript
+         async handleTeamsConfigFetch(_context, _configData) {
+         let response = {};
+
+         response = {
+         config: {
+         type: "auth",
+         suggestedActions: {
+         actions: [
+            {
+               type: "openUrl",
+               value: "https://example.com/auth",
+               title: "Sign in to this app"
+            }]
+         },
+         },
+         };
+
+         return response;
+         }
```

Comment:
Let's make the code have the right intent spaces, this issue applies to other code too